openqa-label-known-issues

```diff
@@ -89,7 +89,7 @@ label_on_issues_from_issue_tracker() {
                 label-on-issue "$id" "$search" "$label" "${after//*\":retry*/1}" "$force_result" && break
             fi
         done
-    )
+    ) || echoerr "$FUNCNAME failed" && exit 1
```

Comment:
no, `label_on_issues_from_issue_tracker` is supposed to "fail", but a failure is nothing fatal here.
failure means it didn't find a matching issue in progress.
In that case it should continue with the other `label_on_issues_*` functions.

Also, if we `exit 1` here, we gained nothing, because the hook script failed and we will get an alert.  It currently does already exit, that's what we want to prevent.

See https://progress.opensuse.org/issues/165716#note-9

Reply:
I was just about to ask similar questions:
1. Why `exit`? That will not make the problem disappear, I would more likely expect something like skipping over problematic items.
2. The `||` part is after a subshell with while loop, that is not likely to fail here. What type of error are you trying to `catch` here?

Reply:
Also note that `$FUNCNAME` is an array.

Reply:
@perlpunk I did some changes. But I am not sure that it is what you expect.

@baierjan 1. that was mistake from my side. I didnt pay attention that the function is in `if` 2. I think the problem is that the `label_on_issues_from_issue_tracker` "fails" and `trap` sees an ERR signal.